lib/plugins/staticFiles.js

```diff
@@ -0,0 +1,131 @@
+ * @param    {Function} [opts.setHeaders] - set custom headers for the Files
+ * (synchronously), The function is called as `fn(res, path, stat)`,
+ * where the arguments are:
+ *      `res` the response object
+ *      `path` the file path that is being sent
+ *      `stat` the stat object of the file that is being sent
+ * @throws   {MethodNotAllowedError}
+ * @throws   {NotAuthorizedError}
+ * @throws   {ResourceNotFoundError}
+ * @returns  {Function} Handler
+ * @example
+ * <caption>
+ * The serveStaticFiles plugin allows you to map a GET route to a
+ * directory on the disk
+ * </caption>
+ * server.get('/public/*', // don't forget the `/*`
+ *      restify.plugins.serveStaticFiles('./documentation/v1')
+ * );
+ * @example
+ * <caption>
+ * The GET `route` and `directory` combination will serve a file
+ * located in `./documentation/v1/index.html` when you attempt to hit
+ * `http://localhost:8080/public/index.html`
+ *
+ * The plugin uses [send](https://github.com/pillarjs/send) under the hood
+ * which is also used by `expressjs` to serve static files. Most of the options
+ * that work with `send` will work with this plugin.
+ *
+ * The default file the plugin looks for is `index.html`
+ * </caption>
+ * server.get('/public/*',
+ *      restify.plugins.serveStaticFiles('./documentation/v1', {
+ *      maxAge: 3600000, // this is in millisecs
+ *      etag: false,
+ *      setHeaders: function setCustomHeaders(response, requestedPath, stat) {
+ *              response.setHeader('restify-plugin-x', 'awesome');
+ *          }
+ *      })
+ * );
+ */
+function serveStaticFiles(directory, opts) {
+    var options = opts || {};
+    // lets assert some options
+    assert.object(options, 'options');
+    assert.string(directory, 'directory');
+    // `send` library relies on `root` to specify the root folder
+    // to look for files
+    options.root = path.resolve(directory);
+
+    return function handleServeStaticFiles(req, res, next) {
+        // Check to make sure that this was either a GET or a HEAD request
+        if (req.method !== 'GET' && req.method !== 'HEAD') {
+            return next(new MethodNotAllowedError('%s', req.method));
+        }
+        // we expect the params to have `*`:
+        // This allows the router to accept any file path
+        var requestedFile = req.params['*'] || 'index.html';
+        // This is used only for sending back correct error message text
+        var requestedFullPath = req.url;
+        // Rely on `send` library to create a stream
+        var stream = send(req, requestedFile, options);
+
+        // Lets handle the various events being emitted by send module
+
+        // stream has ended, must call `next()`
+        stream.on('end', function handleEnd() {
+            return next();
+        });
+
+        // when `send` encounters any `error`, we have the opportunity
+        // to handle the errors here
+        stream.on('error', function handleError(err) {
```

Comment:
how about:

```
        stream.on('error', function handleError(err) {
            // When file does not exist
            if (err.statusCode === 404) {
                return next(new ResourceNotFoundError(requestedFullPath));
            } 
            // or action is forbidden (like requesting a directory)
            return next(new NotAuthorizedError(requestedFullPath));
        });
```

Reply:
will it be always `NotAuthorizedError `?

Reply:
I wanted to put down explicit cases and hence the if/else structure. I foresee we might have other cases(which we find later on as people start using this plugin) where we want to send explicit errors.

If it is fine, I would like to keep the current structure.